### PR TITLE
RTC (DS1307) Kernel Driver + RTC Device Tree + ZRAM (LZ4-LZO) + NOOP IO Sched

### PR DESCRIPTION
**Pull request by JoshDi**
Added RTC (DS1307) Kernel Driver + RTC Device Tree + ZRAM (LZ4-LZO) + IO Scheduler Default=NOOP to ODROID-C2 3.14.y Ubuntu Kernel

### CODE REVIEW

File: arch/arm64/configs/odroidc2_defconfig [hunk 1]
```diff
@@ -138,7 +138,7 @@ CONFIG_RD_LZMA=y
 CONFIG_RD_XZ=y
 CONFIG_RD_LZO=y
 CONFIG_RD_LZ4=y
-# CONFIG_CC_OPTIMIZE_FOR_SIZE is not set
+CONFIG_CC_OPTIMIZE_FOR_SIZE=y
```
Comment (mdrjr) [inline arch/arm64/configs/odroidc2_defconfig:141]: Why?

**Comment by mdrjr** [comment]
I'm not confortable in merging this.. 

Your changes to defconfig doesn't please me.. 
i2c back to builtin
disabled rtl8712 driver
hard-coded your rtc to dt (not everyone has this same rtc)

Can we find a way around this?

**Comment by mlinuxguy** [comment]
One of the patches in my pull request fixes performance when using CFQ scheduler,  might be worth benchmarking before and after my patch to see if it compares to NOOP

**Comment by JoshDi** [comment]
Okay, Ill remove the CONFIG_CC_OPTIMIZE_FOR_SIZE=y line. You dont have to merge my DTS file changes as I agree they are specific for my C2

Also, @mdrjr on my raspberry pi build, they allow for DTS changes via a dtparam in a boot file. Can we replicate this on the C2 kernel? 

dtparam=i2c1=on
dtparam=i2c_arm=on

https://learn.adafruit.com/adafruits-raspberry-pi-lesson-4-gpio-setup/configuring-i2c

File: arch/arm64/configs/odroidc2_defconfig (hunk 1 above)
Comment (JoshDi) [inline arch/arm64/configs/odroidc2_defconfig:141]: Ill remove this optimize for size build option. I agree it is unnecessary

**Pull request closed without merging**